packages/path_provider/path_provider_linux/lib/path_provider_linux.dart

```diff
@@ -12,14 +13,39 @@ import 'package:xdg_directories/xdg_directories.dart' as xdg;
 ///
 /// This class implements the `package:path_provider` functionality for linux
 class PathProviderLinux extends PathProviderPlatform {
+  /// Constructs an instance of [PathProviderLinux]
+  PathProviderLinux() : _environment = Platform.environment;
+
+  /// Constructs an instance of [PathProviderLinux] with the given [environment]
+  @visibleForTesting
+  PathProviderLinux.private({
+    required Map<String, String> environment,
+  }) : _environment = environment;
+
+  final Map<String, String> _environment;
+
   /// Registers this class as the default instance of [PathProviderPlatform]
   static void registerWith() {
     PathProviderPlatform.instance = PathProviderLinux();
   }
 
+  /// Registers this class as the default instance of [PathProviderPlatform]
+  /// with the given [environment]
+  @visibleForTesting
+  static void privateRegisterWith({
+    required Map<String, String> environment,
+  }) {
+    PathProviderPlatform.instance = PathProviderLinux.private(
+      environment: environment,
+    );
+  }
+
   @override
   Future<String?> getTemporaryPath() {
-    return Future<String?>.value('/tmp');
+    final String? tmpdir = _environment['TMPDIR'];
```

Comment:
How about `environmentTmpDir` as the name here, to make its role clearer. This current name doesn't communicate that it doesn't already include the fallback.